From 3e056d403d34f3f4f3d4afa137323accd952a588 Mon Sep 17 00:00:00 2001
From: Xing Zheng <zhengxing@rock-chips.com>
Date: Mon, 23 Apr 2018 21:19:52 +0800
Subject: [PATCH] arm64: dts: rockchip: add bluetooth pcm node for RK3308 EVB
 V10/V11

This patch using PCM rising late1 and slave mode for
Bluetooth HFP.

Change-Id: I4a0188134d7d0ef0690c6c7c9f94fc8ec50c1671
Signed-off-by: Xing Zheng <zhengxing@rock-chips.com>
---
 .../boot/dts/rockchip/rk3308-evb-v10.dtsi     | 21 +++++++++++++++++++
 .../boot/dts/rockchip/rk3308-evb-v11.dtsi     | 21 +++++++++++++++++++
 2 files changed, 42 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3308-evb-v10.dtsi b/arch/arm64/boot/dts/rockchip/rk3308-evb-v10.dtsi
index 30f3cbea0a0d..0e2c72f25259 100644
--- a/arch/arm64/boot/dts/rockchip/rk3308-evb-v10.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3308-evb-v10.dtsi
@@ -111,6 +111,27 @@
 		rockchip,codec = <&acodec>;
 	};
 
+	bluetooth_sound: bluetooth-sound {
+		status = "disabled";
+		compatible = "simple-audio-card";
+		simple-audio-card,format = "dsp_b";
+		simple-audio-card,name = "rockchip,rk3308-pcm";
+		simple-audio-card,mclk-fs = <128>;
+		simple-audio-card,bitclock-inversion = <0>;
+		simple-audio-card,bitclock-master = <&bt_codec>;
+		simple-audio-card,frame-master = <&bt_codec>;
+		simple-audio-card,cpu {
+			sound-dai = <&i2s_2ch_0>;
+		};
+		bt_codec: simple-audio-card,codec {
+			/*
+			 * Using dummy_codec to supports CVSD(8K) and mSBC(16K)
+			 * for HFP.
+			 */
+			sound-dai = <&dummy_codec>;
+		};
+	};
+
 	spdif_tx_sound: spdif-tx-sound {
 		status = "disabled";
 		compatible = "simple-audio-card";
diff --git a/arch/arm64/boot/dts/rockchip/rk3308-evb-v11.dtsi b/arch/arm64/boot/dts/rockchip/rk3308-evb-v11.dtsi
index 1c1dfd9b8ee0..f7785807bf5b 100644
--- a/arch/arm64/boot/dts/rockchip/rk3308-evb-v11.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3308-evb-v11.dtsi
@@ -96,6 +96,27 @@
 		rockchip,codec = <&acodec>;
 	};
 
+	bluetooth_sound: bluetooth-sound {
+		status = "disabled";
+		compatible = "simple-audio-card";
+		simple-audio-card,format = "dsp_b";
+		simple-audio-card,name = "rockchip,rk3308-pcm";
+		simple-audio-card,mclk-fs = <128>;
+		simple-audio-card,bitclock-inversion = <0>;
+		simple-audio-card,bitclock-master = <&bt_codec>;
+		simple-audio-card,frame-master = <&bt_codec>;
+		simple-audio-card,cpu {
+			sound-dai = <&i2s_2ch_0>;
+		};
+		bt_codec: simple-audio-card,codec {
+			/*
+			 * Using dummy_codec to supports CVSD(8K) and mSBC(16K)
+			 * for HFP.
+			 */
+			sound-dai = <&dummy_codec>;
+		};
+	};
+
 	spdif_tx_sound: spdif-tx-sound {
 		status = "disabled";
 		compatible = "simple-audio-card";
-- 
2.35.3

